controllers/tools/applications.js

```diff
@@ -0,0 +1,235 @@
+'use strict';
+const path = require('path');
+const express = require('express');
+const moment = require('moment');
+const concat = require('lodash/concat');
+const groupBy = require('lodash/groupBy');
+const maxBy = require('lodash/maxBy');
+const get = require('lodash/get');
+const mean = require('lodash/mean');
+const uniqBy = require('lodash/uniqBy');
+const minBy = require('lodash/minBy');
+const times = require('lodash/times');
+
+const { PendingApplication, SubmittedApplication } = require('../../db/models');
+const { getDateRange } = require('./helpers');
+
+const router = express.Router();
+
+const DATE_FORMAT = 'YYYY-MM-DD';
+
+function applicationsByDay(responses) {
+    if (responses.length === 0) {
+        return [];
+    }
+
+    const grouped = groupBy(responses, function(response) {
+        return moment(response.createdAt).format(DATE_FORMAT);
+    });
+
+    const newestResponse = maxBy(responses, response => response.createdAt);
+    const oldestResponse = minBy(responses, response => response.createdAt);
+    const oldestResponseDate = moment(oldestResponse.createdAt);
+
+    const daysInRange = moment(newestResponse.createdAt)
+        .startOf('day')
+        .diff(oldestResponseDate.startOf('day'), 'days');
+
+    const dayData = times(daysInRange, function(n) {
+        const key = oldestResponseDate
+            .clone()
+            .add(n, 'days')
+            .format(DATE_FORMAT);
+
+        const responsesForDay = grouped[key] || [];
+
+        return {
+            x: key,
+            y: responsesForDay.length
+        };
+    });
+
+    return dayData;
+}
+
+function minMaxAvg(arr) {
+    const sorted = arr.slice().sort((a, b) => a - b);
+    return {
+        lowest: sorted[0] || 0,
```

Comment:
You could maybe use lodash `min` and `max` to avoid work this out by manually sorting